### ISSUE TITLE
[Feature Request] Choice model for app.context or Choice plugin

### ISSUE THREAD

**Issue by nurdism** [Kong/insomnia#1371]
Something similar to the prompt plugin, but being an enum/array vs string, this could be very useful in many situations.

My use case is I have a REST API, I have a rout to create token, it takes an enum of 'rest' or 'validate' for the token type. Instead of adding a new request or manually inputting 'rest' or 'validate', it would be nice to have a prompt to select one of them.

**Comment by gschier** [comment]
I just remembered there _is_ a way to achieve something similar already. The Prompt dialog supports a `hints` property that can be used to show suggestions below the text input box.

```js
const value = await context.app.prompt('Some Prompt', {
    label: 'My Label',
    submitName: 'Submit',
    cancelable: true,
    hints: ['Option 1', 'Option 2'],
});

console.log('VALUE', value);
```

![image](https://user-images.githubusercontent.com/587576/56839620-30eff400-6838-11e9-8d60-c14bb4911ae3.png)

**Comment by dpkano** [comment]
Could we just expose the the SelectModal throught the app.js the same way PromptModal is?

**Comment by dpkano** [comment]
I have a change in my local branch, @gschier Could I contribute these lines to unblock a good amount of workflows? Do you think adding the SelectModal to be accessed the same way as PromptModal (e.g. context.app.select) fits the interest of the project?

**Comment by dpkano** [comment]
@gschier ? The prompt hints do not give me the possibility to to have a label/id pair to bind with each option, the SelectModel give this possibility.
